src/main/java/org/openrewrite/java/testing/assertj/JUnitAssertInstanceOfToAssertThat.java

```diff
@@ -0,0 +1,94 @@
+/*
+ * Copyright 2024 the original author or authors.
+ * <p>
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * <p>
+ * https://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.openrewrite.java.testing.assertj;
+
+import org.openrewrite.ExecutionContext;
+import org.openrewrite.Preconditions;
+import org.openrewrite.Recipe;
+import org.openrewrite.TreeVisitor;
+import org.openrewrite.java.JavaIsoVisitor;
+import org.openrewrite.java.JavaParser;
+import org.openrewrite.java.JavaTemplate;
+import org.openrewrite.java.MethodMatcher;
+import org.openrewrite.java.search.UsesType;
+import org.openrewrite.java.tree.Expression;
+import org.openrewrite.java.tree.J;
+import org.openrewrite.java.tree.TypeUtils;
+
+import java.util.List;
+
+public class JUnitAssertInstanceOfToAssertThat extends Recipe {
+
+    @Override
+    public String getDisplayName() {
+        return "JUnit `assertInstanceOf` to AssertJ";
+    }
+
+    @Override
+    public String getDescription() {
+        return "Convert JUnit-style `assertInstanceOf()` to AssertJ's `assertThat().isInstanceOf()`.";
+    }
+
+    @Override
+    public TreeVisitor<?, ExecutionContext> getVisitor() {
+        return Preconditions.check(new UsesType<>("org.junit.jupiter.api.Assertions", false),
+                new JavaIsoVisitor<ExecutionContext>() {
+                    private final MethodMatcher JUNIT_ASSERT_NULL_MATCHER = new MethodMatcher("org.junit.jupiter.api.Assertions assertInstanceOf(..)");
+
+                    @Override
+                    public J.MethodInvocation visitMethodInvocation(J.MethodInvocation method, ExecutionContext ctx) {
```

Comment:
We lack a call to `super.visitMethodInvocation` here, and assign that to a variable `md` you use further down. Otherwise you miss out on assertions that might be nested inside other method invocations, as you could have with assertAll, assertSoftly, and other method invocations that could take a lambda.

Reply:
Yeah, you are right. Actually I thought about it, but then I went in line with the other AssertJ recipe's (which don't have the super invocation either). So let's do a follow up PR where we improve the other recipe's as well.